esrally/rally.py

```diff
@@ -822,6 +822,10 @@ def dispatch_sub_command(arg_parser, args, cfg):
             cfg.add(config.Scope.applicationOverride, "driver", "profiling", args.enable_driver_profiling)
             cfg.add(config.Scope.applicationOverride, "driver", "assertions", args.enable_assertions)
             cfg.add(config.Scope.applicationOverride, "driver", "on.error", args.on_error)
+            if args.on_error == "continue-on-non-fatal":
```

Comment:
Since we've settled on targetting `2.2.0`, I feel it's an unnecessary burden to keep this backwards compatibility layer. What if I remove `continue-on-non-fatal`?

Reply:
Works for me.